internal/operator/status.go

```diff
@@ -74,9 +76,7 @@ func (r *StatusReporter) SetRelatedObjects(objs []configv1.ObjectReference) {
 
 // AddRelatedObjects adds to the list of related objects.
 func (r *StatusReporter) AddRelatedObjects(objs []configv1.ObjectReference) {
-	for _, obj := range objs {
-		r.config.RelatedObjects = append(r.config.RelatedObjects, obj)
-	}
+	r.config.RelatedObjects = append(r.config.RelatedObjects, objs...)
```

Comment:
A lot of these changes in the code was because of some `staticcheck` warnings from golangci-lint. There was one warning I haven't fixed yet because I'm not sure how, so I am ignoring it for now, and I hope we can just skip the `lint` test in the CI.

Reply:
Thanks for the tip! Most(?) of the lint changes here I manually changed after reading the error, I didn't do `lint:fix` despite in case it would change something crazy :stuck_out_tongue: